packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -30,25 +31,54 @@ import { ComboBoxInput } from './ComboBoxInput';
 import { ComboBoxInputContainer } from './ComboBoxInputContainer';
 import { ComboBoxMenu } from './ComboBoxMenu';
 import { ComboBoxMenuItem } from './ComboBoxMenuItem';
+import { ComboBoxMultipleContainer } from './ComboBoxMultipleContainer';
 
-export type ComboBoxProps<T> = {
+const isMultipleEnabled = <T,>(
+  variable: (T | undefined) | (T[] | undefined),
+  multiple: boolean,
+): variable is T[] | undefined => {
+  return multiple;
+};
+
+// Type guard for multiple did not work
+const isSingleOnChange = <T,>(
+  onChange:
+    | (((value: T) => void) | undefined)
+    | (((value: T, event: SelectValueChangeEvent) => void) | undefined),
+  multiple: boolean,
+): onChange is ((value: T) => void) | undefined => {
+  return !multiple;
+};
```

Comment:
Can you please explain why you have these functions `isMultipleEnabled` and `isSingleOnChange`? I see the comment that type guard didn't work for `isSingleOnChange`, but I don't understand why you can't use `if (multiple) {...}` and are using `if (isMultipleEnabled(value, multiple) {...}` and `if (isSingleOnChange(onChange, multiple) {...}`.

Reply:
in function
```ts
const isMultipleEnabled = <T,>(
  variable: (T | undefined) | (T[] | undefined),
  multiple: boolean,
): variable is T[] | undefined => {
  return multiple;
};
```
middle part `variable is T[] | undefined` "checks" if variable provided is specific type. 
Eg. `selected` can be number or number[]. If I want to do something only when `multiple` is enabled, I need `selected` to be `number[]`.
So code `if (isMultipleEnabled(selected, multiple)) { doSomethingWith(selected) }` checks if multiple is enabled and also checks if selected type is what I need. I don't need casts `selected as number[]`.

Not sure if make sense. If not, feel free to ping me.

Reply:
Ahhh, that makes sense. Thanks for explaining!